Documentation/design-docs/native-hosting.md

```diff
@@ -306,6 +307,38 @@ Closes a host context.
 * `host_context_handle` - handle to the initialized host context to close.
 
 
+### Loading managed components
+To load managed components from native app directly (not using COM or WinRT) the hosting layer exposes a new runtime helper/delegate `hdt_load_assembly_and_get_function_pointer`. Calling the `hostfxr_get_runtime_delegate(handle, hdt_load_assembly_and_get_function_pointer, &helper)` returns a function pointer to the runtime helper with this signature:
+```C
+int load_assembly_and_get_function_pointer_fn(
+    const char_t *assembly_path,
+    const char_t *type_name,
+    const char_t *method_name,
+    const char_t *delegate_type_name,
+    void         *reserved,
+    /*out*/ void **delegate)
+```
+
+Calling this function will load the specified assembly in isolation (into its own `AssemblyLoadContext`) and it will use `AssemblyDependencyResolver` on it to provide dependency resolution. Once loaded it will find the specified type and method and return a native function pointer to that method. The method's signature can be specified via the delegate type name.
+* `assembly_path` - Path to the assembly to load. In case of complex component, this should be the main assembly of the component (the one with the `.deps.json` next to it). Note that this does not have to be the assembly from which the `type_name` and `method_name` are.
+* `type_name` - Assembly qualified type name to find
+* `method_name` - Name of the method on the `type_name` to find. The method must be `public static` and must match the signature of `delegate_type_name`.
+* `delegate_type_name` - Assembly qualified delegate type name for the method signature, or null. If this is null, the method signature is assumed to be:
+```C#
+public delegate int ComponentEntryPoint(IntPtr args, int sizeBytes);
+```
+This maps to native signature:
+```C
+int component_entry_point_fn(void *arg, int32_t arg_size_in_bytes);
+```
+* `reserved` - parameter reserved for future extensibility, currently unused and must be `0`.
+* `delegate` - out parameter which receives the native function pointer to the requested managed method.
+
+It is allowed to call the returned runtime helper many times for different assemblies or just different methods from the same assembly. It is not required to get the helper every time. The implementation of the helper will cache loaded assemblies, so asking to load the same assembly twice will end up loading it only once and reusing it from that point onward. Ideally components should not take a dependency on this behavior, which means components should not have global state. Global state in components is typically just cause for problems. For example it may create ordering issues or unintended side effects and so on.
```

Comment:
```suggestion
It is allowed to call the returned runtime helper many times for different assemblies or different methods from the same assembly. It is not required to get the helper every time. The implementation of the helper will cache loaded assemblies, so requests to load the same assembly twice will load it only once and reuse it from that point onward. Ideally components should not take a dependency on this behavior, which means components should not have global state. Global state in components is typically just cause for problems. For example it may create ordering issues or unintended side effects and so on.
```